### ISSUE TITLE
"use of unstable library feature 'fs'" warning can't be silenced

### ISSUE THREAD

**Issue by DanB91** [rust-lang/rust#22671]
Even when having #![feature(fs)] in the crate attributes, "use of unstable library feature 'fs'" is not silenced.  All other warnings I have in the crate attributes, like `#![feature(io)]`, are correctly silenced.

I am running using the nightly build as of 2/22/15 on Arch Linux.

My src/lib.rs:

```Rust
#![feature(fs)]
#![feature(io)] 
#![allow(non_snake_case)]
#![allow(dead_code)]
pub mod gb_memory;
```

Warnings from terminal:
```
src/gb_memory.rs:1:5: 1:12 warning: use of unstable library feature 'fs'
src/gb_memory.rs:1 use std::fs;
                       ^~~~~~~
src/gb_memory.rs:1:5: 1:12 help: add #![feature(fs)] to the crate attributes to silence this warning
src/gb_memory.rs:1 use std::fs;
                       ^~~~~~~
src/gb_memory.rs:97:22: 97:36 warning: use of unstable library feature 'fs'
src/gb_memory.rs:97     let mut f = try!(fs::File::open(fileName)); 
                                         ^~~~~~~~~~~~~~
<std macros>:1:1: 6:57 note: in expansion of try!
src/gb_memory.rs:97:17: 97:48 note: expansion site
src/gb_memory.rs:97:22: 97:36 help: add #![feature(fs)] to the crate attributes to silence this warning
src/gb_memory.rs:97     let mut f = try!(fs::File::open(fileName)); 
                                         ^~~~~~~~~~~~~~
<std macros>:1:1: 6:57 note: in expansion of try!
src/gb_memory.rs:97:17: 97:48 note: expansion site

```

**Comment by untitaker** [comment]
Have you tried:

    #![feature(fs,io)]

**Comment by DanB91** [comment]
Tried `#![feature(fs,io)]` and `#![feature(io,fs)]`.  `io` warnings are silenced, but not `fs`.

**Comment by alexcrichton** [comment]
Do you have a standalone example that exhibits this behavior as well?

**Comment by libnc** [comment]
Works well on  rustc 1.0.0-nightly (522d09dfe 2015-02-19) (built 2015-02-21) Windows x64.
```
#![feature(old_io)]
#![feature(fs)]
#![allow(unused_variables)]

use std::fs;
use std::old_io as io;

fn main() {

io::println("Hello");
let x = fs::create_dir("repr");

}
```

**Comment by alexcrichton** [comment]
@libnc that code snippet doesn't generate any warnings for me, does it generate warnings for you?

**Comment by DanB91** [comment]
I apologize for this, but this bug should probably be rejected:

in main.rs I had:
```Rust
#![feature(io)] 
mod gb_memory;
use gb_memory::*;
```

but in lib.rs I had
```Rust
#![feature(fs)]
#![feature(io)] 
#![allow(non_snake_case)]
#![allow(dead_code)]
pub mod gb_memory;
```

So since `#![feature(io)] ` was in both files, it was silenced.  But since `#![feature(fs)]` was only in lib.rs and not main.rs, it was not silenced.  Adding `#![feature(fs)]` silenced the warnings.  Not a huge fan of the fact I had to put a silence directive in two spots, but I'd imagine this is working as designed.

**Comment by libnc** [comment]
@alexcrichton 
it works fine for me as well.

Looks like this can be closed...
